src/Symfony/Component/Serializer/Normalizer/SanitizeDenormalizer.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\HtmlSanitizer\HtmlSanitizerInterface;
+use Symfony\Component\Serializer\Attribute\Sanitize;
+use Symfony\Component\Serializer\Exception\BadMethodCallException;
+use Symfony\Component\Serializer\Exception\InvalidArgumentException;
+use Symfony\Component\Serializer\Exception\LogicException;
+
+/**
+ * Denormalizer that sanitizes string properties marked with the #[Sanitize] attribute.
+ *
+ * @author Mohamed Senoussi <lesfootix@gmail.com>
+ */
+final class SanitizeDenormalizer implements DenormalizerInterface
+{
+    use DenormalizerAwareTrait;
+
+    public function __construct(
+        private readonly HtmlSanitizerInterface $defaultSanitizer,
+        private readonly array $sanitizers = [],
+    ) {}
+
+    public function denormalize(mixed $data, string $type, ?string $format = null, array $context = []): mixed
+    {
+        if (!isset($this->denormalizer)) {
+            throw new BadMethodCallException('Please set a denormalizer before calling denormalize()!');
+        }
+
+        if (!is_array($data) || !class_exists($type)) {
+            return $data;
+        }
+
+        $reflectionClass = new \ReflectionClass($type);
+        foreach ($reflectionClass->getProperties() as $property) {
+            $attributes = $property->getAttributes(Sanitize::class);
+            if ($attributes && array_key_exists($property->getName(), $data)) {
+                /** @var Sanitize $sanitizeAttribute */
+                $sanitizeAttribute = $attributes[0]->newInstance();
+                $sanitizer = $this->getSanitizer($sanitizeAttribute->sanitizer);
+                $data[$property->getName()] = $this->sanitizeValue($data[$property->getName()], $sanitizer, $property, $type);
+            }
+        }
+
+        return $this->denormalizer->denormalize($data, $type, $format, $context);
```

Comment:
If you can the denormalizer with the same type, same format and same context, this will lead to an infinite loop as the Serializer class will resolve the same denormalizer again.